aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,732 @@
+
+.. code-block:: none
+
+  @model TypeNameOfModel<TModel>
+
+For example, if you create a new ASP.NET Core MVC app with individual user accounts, the *Views/Account/Login.cshtml* Razor view file contains the follow model declaration:
+
+.. code-block:: c#
+
+  @model LoginViewModel
+
+In the class example in :ref:`razor-directives-label`, the class generated inherits from ``RazorPage<dynamic>``. By adding an ``@model`` you control what’s inherited. For example
+
+.. code-block:: c#
+
+  @model LoginViewModel
+
+Generates the following class
+
+.. code-block:: c#
+
+ public class _Views_Account_Login_cshtml : RazorPage<LoginViewModel>
+
+This allows you to access the strongly typed model in your Razor page through the ``Model`` property:
+
+.. code-block:: none
+
+  <div>The Login Email: @Model.Email</div>
+  
+Obviously you must pass the model from the controller to the view. See :ref:`strongly-typed-models-keyword-label` for more information.
+
+``@inherits``
+^^^^^^^^^^^^^^^
+
+The ``@inherits`` directive is similar to the model directive. ``@inherits`` gives you full control of the class your Razor page inherits:
+
+.. code-block:: none
+
+ @inherits TypeNameOfClassToInheritFrom
+
+For instance, let’s say we had the following custom Razor page type:
+
+.. literalinclude:: razor/sample/Classes/CustomRazorPage.cs
+  :language: c#
+
+The following Razor would generate ``<div>Custom text: Hello World</div>``.
+
+.. literalinclude:: razor/sample/Views/Home/Contact10.cshtml
+  :language: html
+
+The ``@inherits`` keyword is not allowed on the same page with the ``@model`` statement. You can pass a model as shown with the following Razor page (when passed "Rick@contoso.com" in the model):
+
+.. literalinclude:: razor/sample/Views/Home/Login1.cshtml
+  :language: html
+  :lines: 1-
+
+Generates this HTML:
+
+.. code-block:: none
+
+  <div>The Login Email: Rick@contoso.com</div>
+  <div>Custom text: Hello World.</div>
+
+.. review: Adding the model to _ViewImports is not needed. We don't need it to pass a model.
+
+While you can't use ``@model`` and ``@inherits`` on the same page, you can have ``@model`` in a *_ViewImports.cshtml* file that the Razor page imports. See :doc:`/mvc/views/layout`. For example, if your Razor view imported the following *_ViewImports.cshtml* file:
+
+.. literalinclude:: razor/sample/Views/_ViewImportsModel.cshtml
+  :language: html
+
+The following Razor page, when passed "Rick@contoso.com" in the model:
+
+.. literalinclude:: razor/sample/Views/Home/Login2.cshtml
+  :language: html
+
+Generates this HTML markup:
+
+.. code-block:: none
+
+  <div>The Login Email: Rick@contoso.com</div>
+  <div>Custom text: Hello World</div>
+
+
+``@inject``
+^^^^^^^^^^^^^^
+The ``@inject`` directive enables you to inject a service from your :doc:`service container </fundamentals/dependency-injection>`  into your Razor page for use. See :doc:`/mvc/views/dependency-injection`.
```

Comment:
The example below of replacing the `Html` property with the `IHostingEnvironment` isn't something you would actually do. Instead might replace the `Html` property with your own set of HTML helpers. 

The overriding semantics are also a bit hard to follow as explained. Instead of dealing with this in this PR can we open an issue to get this content added to the [Injecting Services into Views](https://docs.asp.net/en/latest/mvc/views/dependency-injection.html) article and remove this content for now?